src/solver/sideWPointSelect.ml

```diff
@@ -0,0 +1,185 @@
+  end
+module Sides : S =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  functor (VS:Set.S with type elt = S.v) ->
+  struct
+    type data = unit
+
+    let create_data _ _ = ()
+    let notify_side _ _ _ = ()
+    let record_destabilized_vs = false
+    let veto_widen state called old_sides x y = false
+    let should_mark_wpoint state called old_sides x y _ = match x with | Some(x) -> VS.mem x old_sides | None -> true
+  end
+(* TODO: The following two don't check if a vs got destabilized which may be a problem. *)
+module UnstableSelf : S =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  functor (VS:Set.S with type elt = S.v) ->
+  struct
+    type data = { is_stable: S.v -> bool; add_infl: S.v -> S.v -> unit }
+
+    let create_data is_stable add_infl = { is_stable; add_infl }
+    let notify_side data x y = (match x with None -> () | Some x -> data.add_infl x y)
+    let record_destabilized_vs = false
+    let veto_widen _ _ _ _ _ = false
+    (* TODO test/remove. Side to y destabilized itself via some infl-cycle. The above add_infl is only required for this option. Check for which examples this is problematic! *)
+    let should_mark_wpoint state called old_sides x y _ = not (state.is_stable y) (* this is very expensive since it folds over called! see https://github.com/goblint/analyzer/issues/265#issuecomment-880748636 *)
+  end
+module UnstableCalled : S =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  functor (VS:Set.S with type elt = S.v) ->
+  struct
+    type data = { is_stable: S.v -> bool }
+
+    let create_data is_stable _ = { is_stable }
+    let notify_side _ _ _ = ()
+    let record_destabilized_vs = false
+    let veto_widen state called old_sides y x = false
+    (* TODO test/remove. Widen if any called var (not just y) is no longer stable. Expensive! *)
+    let should_mark_wpoint state called old_sides y x _ = HM.fold (fun k _ a -> a || not (state.is_stable k)) called false
+  end
+module Cycle : S =
+  functor (S:EqConstrSys) ->
+  functor (HM:Hashtbl.S with type key = S.v) ->
+  functor (VS:Set.S with type elt = S.v) ->
+  struct
+    type data = unit
+
+    let create_data _ _ = ()
+    let notify_side _ _ _ = ()
+    let record_destabilized_vs = true
+    let veto_widen state called old_sides x y = false
+    (* destabilized a called or start var. Problem: two partial context calls will be precise, but third call will widen the state. *)
+    (* if this side destabilized some of the initial unknowns vs, there may be a side-cycle between vs and we should make y a wpoint *)
+    let should_mark_wpoint state called old_sides x y cycle =
+      match cycle with
+      | Some cycle ->
+        if tracing && cycle then trace "side_widen" "cycle: should mark wpoint %a" S.Var.pretty_trace y;
+        cycle
+      | None ->
+        Logs.warn "destabilize_vs information not provided to side_widen cycle strategy";
```

Comment:
Why would this be the case now? If it can only happen when the implementation itself is buggy, then this warning is useless to the user. An `assert` (or `Option.get`) in this case would be appropriate.